redis/server.cc

```diff
@@ -152,26 +153,57 @@ redis_server::connection::~connection() {
 
 future<> redis_server::connection::process()
 {
-    return do_until([this] {
-        return _read_buf.eof();
-    }, [this] {
-        return with_gate(_pending_requests_gate, [this] {
-            return process_request().then_wrapped([this] (auto f) {
-                try {
-                    f.get();
-                }
-                catch (redis_exception& e) {
-                    write_reply(e);
-                }
-                catch (std::exception& e) {
-                    write_reply(redis_exception { e.what() });
-                }
-                catch (...) {
-                    write_reply(redis_exception { "Unknown exception" });
-                }
+    auto reader = [this] {
+        return do_until([this] { return _read_buf.eof(); }, [this] {
+            return with_gate(_pending_requests_gate, [this] {
+                _parser.init();
+                return _read_buf.consume(_parser).then([this] {
+                    if (_parser.eof()) {
+                        return make_ready_future<>();
+                    }
+                    ++_server._stats._requests_serving;
+                    utils::latency_counter lc;
+                    lc.start();
+                    promise<redis_server::result> pr;
+                    auto fut = pr.get_future();
+                    (void)process_request_internal(std::move(_parser.get_request())).then_wrapped([this, pr = std::move(pr), lc = std::move(lc)] (auto&& fut) mutable {
```

Comment:
"fut" here is different from the "fut" above? It's confusing...

Reply:
Thanks, I got it.